### PR TITLE
Frontier template `isBlockFinalized` & `isTxFinalized` RPC methods

### PR DESCRIPTION
**Pull request by fgamundi**
Ports `isBlockFinalized` & `isTxFinalized` RPC methods from Moonbeam to Frontier container chain template.

### CODE REVIEW

File: container-chains/nodes/frontier/src/rpc/finality.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+}
+
+fn is_block_finalized_inner<B: Block<Hash = H256>, C: HeaderBackend<B> + 'static>(
+    backend: &(dyn fc_api::Backend<B>),
+    client: &C,
+    raw_hash: H256,
+) -> RpcResult<bool> {
+    let substrate_hash =
+        match futures::executor::block_on(frontier_backend_client::load_hash::<B, C>(
```
Comment (tmpolaczyk) [inline container-chains/nodes/frontier/src/rpc/finality.rs:50]: I don't understand why are we using an `async_trait` and `block_on` at the same time, but I see moonbeam does the same so I guess just leave it like this...

File: container-chains/nodes/frontier/src/rpc/finality.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+use sp_runtime::traits::Block;
+use std::{marker::PhantomData, sync::Arc};
+
+#[rpc(server)]
+#[async_trait::async_trait]
+pub trait FrontierFinalityApi {
+    /// Reports whether a Substrate or Ethereum block is finalized.
+    /// Returns false if the block is not found.
+    #[method(name = "frnt_isBlockFinalized")]
+    fn is_block_finalized(&self, block_hash: H256) -> RpcResult<bool>;
```
Comment (tmpolaczyk) [inline container-chains/nodes/frontier/src/rpc/finality.rs:14]: Why are we not adding the tx finalized method?

       fn is_tx_finalized(&self, tx_hash: H256) -> RpcResult<bool>;

**Review by tmpolaczyk — APPROVED** [review]

File: container-chains/nodes/frontier/src/rpc/finality.rs (hunk 2 above)
Comment (girazoki) [reply container-chains/nodes/frontier/src/rpc/finality.rs:14]: yes let's add this, this is fairly important

File: container-chains/nodes/frontier/src/rpc/finality.rs (hunk 1 above)
Comment (fgamundi) [reply container-chains/nodes/frontier/src/rpc/finality.rs:50]: I was kinda wondering the same thing. It seems to work fine without the `async_trait` decorator. Maybe it's ok to remove it?

Comment (tmpolaczyk) [reply container-chains/nodes/frontier/src/rpc/finality.rs:70]: Actually the proper solution is to turn the `fn` into `async fn`, and `block_on` into `await`, but maybe that doesn't work for some reason

Comment (fgamundi) [reply container-chains/nodes/frontier/src/rpc/finality.rs]: Updated the code. Pls take a look

Comment (tmpolaczyk) [reply container-chains/nodes/frontier/src/rpc/finality.rs]: Perfect, you can propose a similar change to moonbeam

**Review by tmpolaczyk — APPROVED** [review]

File: container-chains/nodes/frontier/src/rpc/finality.rs (hunk 1 above)
Comment (fgamundi) [reply container-chains/nodes/frontier/src/rpc/finality.rs]: https://github.com/moonbeam-foundation/moonbeam/pull/2784

**Pull request merged**